modules/logging-loki-storage.adoc

```diff
@@ -0,0 +1,249 @@
+
+.Procedure
+
+. Copy the service account credentials received from GCP into a file name `key.json`.
+
+. Create an Object Storage secret with keys `bucketname` and `key.json` as follows:
+
++
+[source,terminal]
+----
+oc create secret generic logging-loki-gcs \
+  --from-literal=bucketname="<BUCKET_NAME>" \
+  --from-file=key.json="<PATH/TO/KEY.JSON>"
+----
+
+Where `logging-loki-gcs` is the secret name, `<BUCKET_NAME>` is the name of bucket created in prerequisites step and `<PATH/TO/KEY.JSON>` is the file path where the `key.json` was copied to.
+
+** Suggested secret name: `logging-loki-gcs`
+
+** Secret type: `gcs`
+
+
+== Minio
+
+.Prerequisites
+
+* Loki Operator deployed
+
+* link:https://operator.min.io/[Minio] deployed on your Cluster.
+
+* Create a link:https://docs.min.io/docs/minio-client-complete-guide.html[bucket] on Minio via CLI.
+
+.Procedure
+
+* Create an Object Storage secret with keys as follows:
+
+[source,terminal]
+----
+oc create secret generic logging-loki-minio \
+  --from-literal=bucketnames="<BUCKET_NAME>" \
+  --from-literal=endpoint="<MINIO_BUCKET_ENDPOINT>" \
+  --from-literal=access_key_id="<MINIO_ACCESS_KEY_ID>" \
+  --from-literal=access_key_secret="<MINIO_ACCESS_KEY_SECRET>"
+----
+
+** Suggested secret name: `logging-loki-minio`
+
+** Secret type: `s3`
+
+
+== OpenShift Data Foundation
+
+.Prerequisites
+
+* Loki Operator deployed
+
+* Deploy the [OpenShift Data Foundation](https://access.redhat.com/documentation/en-us/red_hat_openshift_data_foundation/4.12) on your cluster.
+
+.Procedure
+
+. Create an ObjectBucketClaim in `openshift-logging` namespace:
++
+[source,yaml]
+----
+apiVersion: objectbucket.io/v1alpha1
+kind: ObjectBucketClaim
+metadata:
+  name: loki-bucket-odf
+  namespace: openshift-logging
+spec:
+  generateBucketName: loki-bucket-odf
+----
+
+. Get bucket properties from the associated ConfigMap:
++
+[source,terminal]
+----
+BUCKET_HOST=$(oc get -n openshift-logging configmap loki-bucket-odf -o jsonpath='{.data.BUCKET_HOST}')
+BUCKET_NAME=$(oc get -n openshift-logging configmap loki-bucket-odf -o jsonpath='{.data.BUCKET_NAME}')
+BUCKET_PORT=$(oc get -n openshift-logging configmap loki-bucket-odf -o jsonpath='{.data.BUCKET_PORT}')
+----
+
+. Get bucket access key from the associated Secret:
++
+[source,terminal]
+----
+ACCESS_KEY_ID=$(oc get -n openshift-logging secret loki-bucket-odf -o jsonpath='{.data.AWS_ACCESS_KEY_ID}' | base64 -d)
+SECRET_ACCESS_KEY=$(oc get -n openshift-logging secret loki-bucket-odf -o jsonpath='{.data.AWS_SECRET_ACCESS_KEY}' | base64 -d)
+----
+
+.. Create an Object Storage secret with keys as follows:
++
+[source,terminal]
+----
+oc create -n openshift-logging secret generic logging-loki-odf \
+--from-literal=access_key_id="${ACCESS_KEY_ID}" \
+--from-literal=access_key_secret="${SECRET_ACCESS_KEY}" \
+--from-literal=bucketnames="${BUCKET_NAME}" \
+--from-literal=endpoint="https://${BUCKET_HOST}:${BUCKET_PORT}"
```

Comment:
One more thing we should mention here is: if the schema of endpoint is `https`, users should:
1. create a configmap with below yaml in the namespace where the lokistack will be in(here it's `openshift-logging`):
```
cat << EOF | oc create -f -
apiVersion: v1
kind: ConfigMap
metadata:
  name: lokistack-odf-ca
  namespace: openshift-logging
  annotations:
    service.beta.openshift.io/inject-cabundle: "true"
EOF
```
note: the configmap name can be changed.
2. users should add below lines into the lokistack CR:
```
spec:
  storage:
    secret:
      name: logging-loki-odf
      type: s3
    tls:
      caName: lokistack-odf-ca
```
Or the loki pods may not be ready and raise some `x509` errors.

If the schema is `http`, then no need to do above steps.